aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -325,6 +339,13 @@ module aptos_framework::delegation_pool {
         effective_after_secs: u64,
     }
 
+    /// Tracks a delegation pool's allowlist of delegators.
+    /// If allowlisting is enabled, existing delegators are not implicitly allowlisted and they can be individually
+    /// evicted later by the pool owner.
+    struct DelegationPoolAllowlisting has key {
```

Comment:
Is it possible to move this functionality into a separate module? In 2023, there have been enough improvements to Move and the Aptos Framework to allow for much more modular designs that how this and the stake module were written in 2022 + early 2023.

Reply:
I agree, the allowlist resource and events could be moved to a separate module.
`assert_delegator_allowlisted` can be entirely implemented there and be externally available for `add_stake` and `reactivate_stake`.
However, operations that require the pool's ownership capability (especially `evict_delegator`) would have to be partially implemented in `delegation_pool.move` and forward the request to `delegation_pool_allowlist.move` to update the allowlist.
Validating the ownership entirely in `delegation_pool_allowlist.move` would require importing `delegation_pool.move` which becomes a circular dependency. We could save the owner address in `DelegationPoolAllowlisting` when enabling allowlisting and then update it if `DelegationPoolOwnership` should ever be transferable.

What would be the recent improvements of the Aptos Framework that could help us with this separation?